### PR TITLE
new code to process delta files

### PR DESCRIPTION
**Pull request by dho937**

### CODE REVIEW

**Comment by dho937** [comment]
@karlmdavis  This is my initial approach to processing the delta files.  I still need to do more unit testing but wanted to give you a chance to look at first.  Thanks.

File: bluebutton-data-pipeline-rif-load/src/main/java/gov/hhs/cms/bluebutton/data/pipeline/rif/load/RifLoader.java [hunk 1]
```diff
@@ -444,12 +444,19 @@ private void processAsync(List<RifRecordEvent<?>> recordsBatch, PostgreSqlCopyIn
...
-						entityManager.persist(record);
+					} else if (strategy == LoadStrategy.INSERT_UPDATE_NON_IDEMPOTENT) {
+						if (rifRecordEvent.getRecordAction().equals(RecordAction.INSERT)) {
+							loadAction = LoadAction.INSERTED;
+							entityManager.persist(record);
+						} else {
+							loadAction = LoadAction.UPDATED;
+							entityManager.merge(record);
+						}
+
```
Comment (karlmdavis) [inline bluebutton-data-pipeline-rif-load/src/main/java/gov/hhs/cms/bluebutton/data/pipeline/rif/load/RifLoader.java]: That block is an indented mess, but that's not your fault. Can you just add a `// TODO: refactor this to be less of an indented mess` comment, which we can address later?

File: bluebutton-data-pipeline-rif-load/src/test/java/gov/hhs/cms/bluebutton/data/pipeline/rif/load/RifLoaderIT.java [hunk 2]
```diff
@@ -93,9 +103,16 @@ private void loadSample(StaticRifResourceGroup sampleGroup) {
...
+		/*
+		 * Assert.assertEquals("Unexpected number of successful records: " +
+		 * successfulRecords, sampleResources.stream().mapToInt(r ->
+		 * r.getRecordCount()).sum(), successfulRecords.stream().filter(r ->
+		 * r.getLoadAction() == LoadAction.INSERTED).count());
+		 * Assert.assertEquals("Unexpected number of successful records: " +
+		 * successfulRecords, sampleResources.stream().mapToInt(r ->
+		 * r.getRecordCount()).sum(), successfulRecords.stream().filter(r ->
+		 * r.getLoadAction() == LoadAction.UPDATED).count());
+		 */
```
Comment (karlmdavis) [inline bluebutton-data-pipeline-rif-load/src/test/java/gov/hhs/cms/bluebutton/data/pipeline/rif/load/RifLoaderIT.java]: I don't think we can merge this unless/until we get the test coverage in place. Probably need to expand it a bit, too, as there is more that can go wrong now.

**Comment by dho937** [comment]
@karlmdavis  Created some more unit tests for our RIF "updates" to our sample files.  Please review when you get a chance.  Thanks.

**Pull request merged**